### DISCUSSION TITLE
Custom Doc Tag Page Content (per Tag)

### DISCUSSION THREAD

**Discussion by niklasp** [facebook/docusaurus#8800] [Q&A]
I have to ask the question again here, because I cannot yet find a way to do it, even with looking at Docusaurus Code and asking on stackoverflow and discord:

My aim is to generate a separate and unique page for each doc tag. Currently I have swizzled the `DocTagDocListPage` and understand that by changing that component, I can change the view to the user.

I do however want, that each tag, e.g. `/docs/tags/contribute` has a different content than `/docs/tags/help`. That content should be loaded from a .mdx file e.g. `/docs/tagsPages/contribute.mdx`. When there is no `.mdx` file, the default DocTagDocListPage should be shown.

How can I achieve that? Currently I have a hacky workaround by importing `.mdx` content directly and then displaying it in the page. However this of course e.g. breaks the edit link (tag pages have no edit link).

`DocTagDocListPage/index.js`
```js
import Contribute from '/docs/category/contribute.mdx';

  function TagPageContent() {
    switch( tag.label ) {
      ...
      case 'contribute':
        return <Contribute/>
      ...
      default:
        return <></>
    }
  }
```

Then I would add that `TagPageContent` in the render function.

Any help is very much appreciated for a community doc platform I am trying to build. If anything is unclear I can clarify.

**Comment by slorber** [comment]
Unfortunately, we don't support creating custom tag-related pages.

Your workaround is probably the best solution although it is not very scalable.

Another solution could be to try to extend the docs plugin to get it to produce extra tag routes in the `contentLoaded` lifecycle. This would require a bit of retro-engineering to implement 🤪 Related to: https://github.com/facebook/docusaurus/issues/4138

---

That's an interesting feature request to implement IMHO, but not sure how it should be designed.

I think we should have a `tags.yml` similar to what we do for blog `authors.yml`: https://docusaurus.io/docs/blog#global-authors

And should be possible to reference some custom MDX content for the intro of a tag page?
